src/registration.rs

```diff
@@ -138,115 +138,130 @@ impl Registrar {
-                                                          0xc0a8,0x4),
-                                        netmask: Ipv6Addr::new(0xffff,0xffff,
-                                                               0xffff,0xffff,
-                                                               0xffff,0xffff,
-                                                               0xffff,0xffff),
-                                        broadcast: None })
-        },
-        Interface {
-            name: "eth0".to_owned(),
-            addr: IfAddr::V4(Ifv4Addr { ip: Ipv4Addr::new(192,168,0,4),
-                                        netmask: Ipv4Addr::new(255,255,255,0),
-                                        broadcast: Some(
-                                            Ipv4Addr::new(192,160,0,255)) })
-        },
-        ];
-    // we should get eth0 in IPv4
-    let ip_addr = registrar.get_ip_addr_from_ifaces(&ifaces2, &None);
-    assert!(ip_addr == Some("192.168.0.4".to_owned()));
+    }
+
+    describe! ipv6 {
+        before_each {
+            use super::super::get_if_addrs::*;
+            use std::net::{ Ipv4Addr, Ipv6Addr };
+
+            let interfaces: Vec<Interface> = vec![
+                Interface {
+                    name: "eth0".to_owned(),
+                    addr: IfAddr::V6(Ifv6Addr {
+                        ip: Ipv6Addr::new(0,0,0,0,0,0xffff,0xc0a8,0x4),
+                        netmask: Ipv6Addr::new(0xffff,0xffff,0xffff,0xffff,
+                                               0xffff,0xffff,0xffff,0xffff),
+                        broadcast: None
+                    })
+                },
+                Interface {
+                    name: "eth0".to_owned(),
+                    addr: IfAddr::V4(Ifv4Addr {
+                        ip: Ipv4Addr::new(192,168,0,4),
+                        netmask: Ipv4Addr::new(255,255,255,0),
+                        broadcast: Some(Ipv4Addr::new(192,160,0,255))
+                    })
+                },
+                Interface {
+                    name: "eth1".to_owned(),
+                    addr: IfAddr::V6(Ifv6Addr {
+                        ip: Ipv6Addr::new(0,0,0,0,0,0xffff,0xc0a8,0x4),
+                        netmask: Ipv6Addr::new(0xffff,0xffff,0xffff,0xffff,
+                                               0xffff,0xffff,0xffff,0xffff),
+                        broadcast: None
+                    })
+                }
+            ];
+        }
+
+        it "should return IPv6" {
+            let ip_addr = registrar.get_ip_addr_from_ifaces(&interfaces, &Some("eth1".to_owned()))
+                .unwrap();
+            assert_eq!(ip_addr, "::ffff:192.168.0.4");
+        }
+
+        it "should return IPv4 if both are specified" {
+            let ip_addr = registrar.get_ip_addr_from_ifaces(&interfaces, &Some("eth0".to_owned()))
+                .unwrap();
+            assert_eq!(ip_addr, "192.168.0.4");
+        }
+    }
```

Comment:
I'm not too sure. Isn't that already covered by the combination of these 2 cases:
* ["should default to eth0"](https://github.com/fxbox/foxbox/pull/103/files#diff-81c55a5b1db198d473ae06791f9953ffR198 )
* and ["should return IPv4 if both are specified"](https://github.com/fxbox/foxbox/pull/103/files#diff-81c55a5b1db198d473ae06791f9953ffR261)